app/controllers/concerns/votes_controller.rb

```diff
@@ -0,0 +1,16 @@
+class VotesController < ApplicationController
+  before_action :authenticate_user!
+
+  def create
+    review = Review.find(params[:review_id])
+    type = params[:type]
+    if review.update_vote(current_user, type)
+      flash[:success] = "#{type} saved."
+    else
+      flash[:alert] = "Sorry, you're only allowed one #{type} per review."
+    end
+
+    redirect_to restaurant_path(review.restaurant)
+  end
+
```

Comment:
Here I went with adding a param[:type] string of either "Upvote" / "Downvote"  - I found that it would make the success/error messages a little easier to create, and because the review model does string matching internally to figure out if it should increment/decrement, this implementation seemed secure enough. Let me know what you think.